force-app/main/default/classes/OpportunityHelper.cls

```diff
@@ -0,0 +1,149 @@
+        for(Opportunity opp : oppList){
+            if(opp.Amount <= 5000){
+                opp.addError('Opportunity amount must be greater than 5000');
+                        
+            }
+
+        }
+    }
+
+    public static void setPrimaryContact(List<Opportunity> oppList){
+        List<Id> accountIds = new List<Id>();
+            for (Opportunity opp : oppList) {
+                if (opp.AccountId != null) {
+                    accountIds.add(opp.AccountId);
+                }
+            }
+            
+            Map<Id, Contact> ceoContactsByAccount = new Map<Id, Contact> ([SELECT Id, AccountId, Title FROM Contact WHERE AccountId IN :accountIds AND Title = 'CEO']);
+           
+            Map<Id, Contact> accountIdToContact = new Map<Id, Contact>();
+        
+                for (Contact cont : ceoContactsByAccount.values()) {
+                    accountIdToContact.put(cont.AccountId, cont);
+                }
+                
+            for(Opportunity opp : oppList){
+                if(opp.Primary_Contact__c == null && accountIdToContact.containsKey(opp.AccountId)){
+                    opp.Primary_Contact__c = accountIdToContact.get(opp.AccountId).Id;
+                }
+            }
+        }
+
+    public static void updateDescriptionFromStage(List<Opportunity> oppList){
+
+        system.debug(oppList);    
+        for(Opportunity opp : oppList){
+            if(opp.stageName != null){
+                system.debug(opp.Description);
+                opp.Description += 'Stage Change:' + opp.StageName;
+            }
+        }
+    }
+    
+    public static void setDefaultOpportunityType(List<Opportunity> oppList){
+        for(Opportunity opp : oppList){
+            if (opp.Type == null){
+                opp.Type = 'New Customer';
+                }        
+            } 
+    }
+
+    public static void insertTask(List<Opportunity> oppList){
+        List<Task> tskList = new List<Task>();
+        for (Opportunity opp : oppList){
+            Task tsk = new Task();
+            tsk.Subject = 'Call Primary Contact';
+            tsk.WhatId = opp.Id;
+            tsk.WhoId = opp.Primary_Contact__c;
+            tsk.OwnerId = opp.OwnerId;
+            tsk.ActivityDate = Date.today().addDays(3);
+            tskList.add(tsk);
+        }
+        insert tskList;
+    
+    }
+
+    public static void deleteCloseWonOpportunity(List<Opportunity> oppList){
+        List<Id> accIdList = new List<Id>();
+        for(Opportunity oldOpp : oppList){
+            accIdList.add(oldOpp.AccountId);
+        }
+        Map<Id, Account> accounts = new Map<Id, Account>([SELECT Id, Industry FROM Account WHERE Id IN :accIdList]);
+        for (Opportunity oldOpp : oppList){
+            if (oldOpp.StageName == 'Closed Won' && accounts.get(oldOpp.AccountId).Industry == 'Banking'){
+                oldOpp.addError('Cannot delete closed opportunity');
+            }
+
+       }
+    }
```

Comment:
Do you still need to check this if there is another before delete check that prevents deleting any closed opportunities?